pkg/secrets/service/manager.go

```diff
@@ -37,12 +38,27 @@ const Type = "service"
 
 // serviceCrypter is an encrypter/decrypter that uses the Pulumi servce to encrypt/decrypt a stack's secrets.
 type serviceCrypter struct {
-	client *client.Client
-	stack  client.StackIdentifier
+	client                  *client.Client
+	stack                   client.StackIdentifier
+	supportsBatchEncryption *promise.Promise[bool]
 }
 
 func newServiceCrypter(client *client.Client, stack client.StackIdentifier) config.Crypter {
-	return &serviceCrypter{client: client, stack: stack}
+	return &serviceCrypter{
+		client: client,
+		stack:  stack,
+		supportsBatchEncryption: promise.Run(func() (bool, error) {
+			capabilitiesResponse, err := client.GetCapabilities(context.Background())
+			if err != nil {
+				return false, err
```

Comment:
Should we log these errors to debug logs at least? We never look at the promise result error